rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp

```diff
@@ -376,7 +377,7 @@ bool TypeSupport<MembersType>::serializeROSmessage(
 
   for (uint32_t i = 0; i < members->member_count_; ++i) {
     const auto member = members->members_ + i;
-    void * field = (char *)ros_message + member->offset_;
+    void * field = const_cast<char *>(static_cast<const char *>(ros_message)) + member->offset_;
```

Comment:
Also, I could not convert the `const void *` directly to `char *` so I did it in two steps...